packages/repack/src/webpack/Compiler.ts

```diff
@@ -158,39 +158,47 @@ export class Compiler extends EventEmitter {
     return worker;
   }
 
+  private addProgressSender(platform: string, callback?: SendProgress) {
+    if (!callback) return;
+    this.progressSenders[platform] = this.progressSenders[platform] ?? [];
+    this.progressSenders[platform].push(callback);
+  }
+
+  private removeProgressSender(platform: string, callback?: SendProgress) {
+    if (!callback) return;
+    this.progressSenders[platform] = this.progressSenders[platform].filter(
+      (item) => item !== callback
+    );
+  }
+
   async getAsset(
     filename: string,
     platform: string,
     sendProgress?: SendProgress
   ): Promise<Asset> {
     // Return file from assetsCache if exists
     const fileFromCache = this.assetsCache[platform]?.[filename];
-    if (fileFromCache) {
-      return fileFromCache;
-    }
+    if (fileFromCache) return fileFromCache;
+
+    this.addProgressSender(platform, sendProgress);
 
     // Spawn new worker if not already running
     if (!this.workers[platform]) {
       this.workers[platform] = this.spawnWorker(platform);
     } else if (!this.isCompilationInProgress[platform]) {
+      this.removeProgressSender(platform, sendProgress);
```

Comment:
Is it possible to have a registered progress sender if the compilation is **not** in progress?

Reply:
Oh, right, that makes sense! 🎯